libportability-gfx/src/impls.rs

```diff
@@ -2417,23 +2415,41 @@ pub extern "C" fn gfxAllocateDescriptorSets(
     let sets = unsafe {
         slice::from_raw_parts_mut(pDescriptorSets, info.descriptorSetCount as _)
     };
-    for (set, raw_set) in sets.iter_mut().zip(descriptor_sets.into_iter()) {
-        *set = match raw_set {
-            Ok(set) => Handle::new(set),
-            Err(e) => return match e {
-                pso::AllocationError::OutOfHostMemory => VkResult::VK_ERROR_OUT_OF_HOST_MEMORY,
-                pso::AllocationError::OutOfDeviceMemory => VkResult::VK_ERROR_OUT_OF_DEVICE_MEMORY,
-                pso::AllocationError::OutOfPoolMemory => VkResult::VK_ERROR_OUT_OF_POOL_MEMORY_KHR,
-                pso::AllocationError::IncompatibleLayout => VkResult::VK_ERROR_DEVICE_LOST,
-                pso::AllocationError::FragmentedPool => VkResult::VK_ERROR_FRAGMENTED_POOL,
-            },
+    let mut result = VkResult::VK_SUCCESS;
+    assert_eq!(descriptor_sets.len(), info.descriptorSetCount as usize);
+
+    for (i, raw_set) in descriptor_sets.into_iter().enumerate() {
+        match raw_set {
+            Ok(set) => {
+                sets[i] = Handle::new(set);
+            }
+            Err(e) => {
+                // revert all the changes!
+                for set in sets[..i].iter_mut() {
```

Comment:
What is with the other remaining sets?

Reply:
in the failure case, the other sets can technically be untouched. There is a `for` loop following here that just zeroes them out, for the piece of mind